grpc-server/template_test.go

```diff
@@ -0,0 +1,75 @@
+package grpcserver
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tinkerbell/tink/db/mock"
+	pb "github.com/tinkerbell/tink/protos/template"
+)
+
+const (
+	template1 = `version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world"
+    worker: "{{.device_1}}"
+    actions:
+    - name: "hello_world"
+      image: hello-world
+      timeout: 60`
+
+	template2 = `version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world again"
+    worker: "{{.device_1}}"
+    actions:
+  	- name: "hello_world_again"
+      image: hello-world
+      timeout: 60`
+)
+
+func TestDuplicateTemplateName(t *testing.T) {
```

Comment:
Can you call it `TestCreateTemplate`? and the table test has to work like this imho:
Table Test is not only for one specific use case, but it has to be flexible enough to allow all the one you can think about for the function you are testing, in our case `CreateTemplate`

```golang
// you don't need a testCases struct just use a []testCase
struct testCase type {
    db mock.DB
    name string
    input []string // or templates
    expect func() (*template.CreateResponse, error)
}
```

In order to figure out if the TableTest is well written can you add three other testcases please?

1. Well formatted and successful template creation (only one)
2. A random validation error if we have any
3. Create 2 templates with a different name, both should work

Thanks

Reply:
sure, I will try to add couple of more test cases as you suggested.

Reply:
@gianarb  As mentioned by you I have modified the test struct ( not exactly as you suggested ) so that in future more tests can be added. And I have also added couple of more test cases. But I was not able to test for validation of template because we don't check the validation of a template in `CreateTemplate` function. Template is being validated while creating a workflow here https://github.com/tinkerbell/tink/blob/master/db/workflow.go#L98.